fecgov/openFEC
# Fix incorrect totals for C00508416 (FRIENDS OF JOHN DELANEY) on committee profile page

## Summary (required)

- Resolves #[_3536_](https://github.com/fecgov/openFEC/issues/3656)

_Committee C00508416 filed both F3P and F3 for 2018.  In ofec_totals_combined_mv, this committee has two rows for each form type.  However the mv feeding into this committee's profile page only has F3P data because the committee type is 'P'.  An aggregated view is created based on ofec_totals_combined_mv group by committee and cycle._

_This vw can be used for committees other than committee_type='P' in the future when calculating their totals. Thus the `totals_schema_map` will not be needed._
## How to test the changes locally

1)  Checkout 'feature/fix-delaney-committee-profile' to local
2) A tmp version of the view (ofec_committee_totals_per_cycle_vw_hc) had been created in DEV database 
    run this query to compare total receipts & total disbursements
    
    select * from ofec_committee_totals_per_cycle_vw_hc where committee_id = 'C00508416' and cycle 
    = 2018

    The value should be:  
    total receipts: 6,145,779.44
    total disbursements: 5,899,574.20

3) In the model file totals.py, change the following line to point to the tmp view
    class CommitteeTotalsPerCycle(CommitteeTotals):
        **__tablename__ = 'ofec_committee_totals_per_cycle_vw_hc'**
4) start local server, and run this endpoint
    http://127.0.0.1:5000/v1/committee/C00508416/totals/?sort_null_only=false&sort_hide_null=false&per_page=20&sort=-cycle&page=1&sort_nulls_last=false
  
    Compare the endpoint output with Delaney's candidate profile page:
https://www.fec.gov/data/candidate/P00006213/?cycle=2018&election_full=false

5) To test migration file ( V0127 ), go to local flyway directory,  run this command:
     flyway migrate

## Impacted areas of the application
List general components of the application that this PR will affect:

financial:
/committee/{committee_id}/totals/
